staging/src/k8s.io/client-go/tools/clientcmd/config.go

```diff
@@ -169,15 +169,23 @@ func NewDefaultPathOptions() *PathOptions {
 // modified element.
 func ModifyConfig(configAccess ConfigAccess, newConfig clientcmdapi.Config, relativizePaths bool) error {
 	if UseModifyConfigLock {
-		possibleSources := configAccess.GetLoadingPrecedence()
-		// sort the possible kubeconfig files so we always "lock" in the same order
-		// to avoid deadlock (note: this can fail w/ symlinks, but... come on).
-		sort.Strings(possibleSources)
-		for _, filename := range possibleSources {
+		if configAccess.IsExplicitFile() {
```

Comment:
Is this check really sufficient to ensure that none of the other files will be read / written to below?
I don't think it is.

Why should possible sources contain files we won't actually read|write if that is the case? This seems brittle.

Reply:
Only `writeCurrentContext` and `writePreferences` use `GetLoadingPrecedence `, and `IsExplicitFile` is also used first

https://github.com/kubernetes/kubernetes/blob/1822523afb68f33df242b1c43590154f1a361784/staging/src/k8s.io/client-go/tools/clientcmd/config.go#L196-L206


https://github.com/kubernetes/kubernetes/blob/1822523afb68f33df242b1c43590154f1a361784/staging/src/k8s.io/client-go/tools/clientcmd/config.go#L382-L407
https://github.com/kubernetes/kubernetes/blob/1822523afb68f33df242b1c43590154f1a361784/staging/src/k8s.io/client-go/tools/clientcmd/config.go#L444-L465